### PR TITLE
NIC: Bridged VLAN support

### PR DESCRIPTION
**Pull request by tomponline**
Adds support for `vlan` and `vlan.tagged` settings on `bridged` NIC devices.

`vlan` sets the untagged VLAN ID for the interface, and `vlan.tagged` accepts a comma-delimited list of tagged VLANs to join.

`vlan` accepts a special `none` value which means that it will be removed from the bridge's default untagged VLAN (as joining the bridge's default untagged behaviour is what happens if `vlan` is unspecified).

### CODE REVIEW

**Comment by tomponline** [comment]
jenkins: test this please

File: lxd/device/nic_bridged.go [hunk 1]
```diff
@@ -1188,3 +1194,57 @@ func (d *nicBridged) networkDHCPv6CreateIAAddress(IP net.IP) []byte {
 	binary.BigEndian.PutUint32(data[24:28], uint32(0))                    // Valid lifetime
 	return data
 }
+
+// setupBridgePortVLANs configures the bridge port with the specified VLAN settings in device config.
+func (d *nicBridged) setupBridgePortVLANs(hostName string) error {
+	// Enable vlan_filtering on bridge if needed.
```
Comment (stgraber) [inline lxd/device/nic_bridged.go:1200]: Comment is inaccurate, we're just checking

File: lxd/network/network_utils.go [hunk 2]
```diff
@@ -849,3 +851,43 @@ func UsesIPv6Firewall(netConfig map[string]string) bool {
...
+	content, err := ioutil.ReadFile(fmt.Sprintf("/sys/class/net/%s/bridge/vlan_filtering", interfaceName))
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed getting bridge VLAN status for %q", interfaceName)
+	}
+
+	return strings.TrimSpace(fmt.Sprintf("%s", content)), nil
+}
+
+// BridgeVLANFilterSetStatus sets the status of VLAN filtering on a bridge interface.
+func BridgeVLANFilterSetStatus(interfaceName string, status string) error {
```
Comment (stgraber) [inline lxd/network/network_utils.go:866]: Isn't this dead code now?

Comment (stgraber) [reply lxd/network/network_utils.go:866]: Ah no, we use it on our own managed bridges, nevermind :)

File: lxd/device/nic_bridged.go (hunk 1 above)
Comment (tomponline) [reply lxd/device/nic_bridged.go:1200]: Will fix in my OVS branch.

**Pull request merged**